src/main/java/omero/client.java

```diff
@@ -536,6 +541,58 @@ public boolean isSecure() {
         return nClient;
     }
 
+    /**
+     * Checks whether the host is a URI, returns the Ice endpoint if so
+     * @param pmap Ice properties
+     * @return The Ice endpoint if host was a URI, or null
+     */
+    private String checkForHosturl(Ice.Properties pmap) {
```

Comment:
I'd suggest making this protected and adding unit tests to show what type of parsing you are expecting.

Reply:
The `omero.client` constructor always runs `checkForHosturl` so there wasn't much point in making it protected and checking it on it's own.